pkg/components/exec_node_base_test.go

```diff
@@ -0,0 +1,320 @@
+package components
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	ytv1 "github.com/ytsaurus/yt-k8s-operator/api/v1"
+	apiProxy "github.com/ytsaurus/yt-k8s-operator/pkg/apiproxy"
+	"github.com/ytsaurus/yt-k8s-operator/pkg/canonize"
+	"github.com/ytsaurus/yt-k8s-operator/pkg/consts"
+	"github.com/ytsaurus/yt-k8s-operator/pkg/labeller"
+	"github.com/ytsaurus/yt-k8s-operator/pkg/ytconfig"
+	"go.ytsaurus.tech/library/go/ptr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	testClusterDomain = "fake.zone"
+
+	testNamespace    = "fake"
+	testYtsaurusName = "test"
+)
+
+var (
+	testLogRotationPeriod int64 = 900000
+	testTotalLogSize            = 10 * int64(1<<30)
+
+	testStorageClassname = "yc-network-hdd"
+
+	testResourceReqs = corev1.ResourceRequirements{
+		Limits: corev1.ResourceList{
+			corev1.ResourceCPU:    *resource.NewQuantity(20, resource.BinarySI),
+			corev1.ResourceMemory: *resource.NewQuantity(5*1024*1024*1024, resource.BinarySI),
+		},
+	}
+	testJobResourceReqs = corev1.ResourceRequirements{
+		Limits: corev1.ResourceList{
+			corev1.ResourceCPU:    *resource.NewQuantity(99, resource.BinarySI),
+			corev1.ResourceMemory: *resource.NewQuantity(99*1024*1024*1024, resource.BinarySI),
+		},
+	}
+
+	testLocationChunkCache = ytv1.LocationSpec{
+		LocationType: "ChunkCache",
+		Path:         "/yt/hdd1/chunk-cache",
+	}
+	testLocationSlots = ytv1.LocationSpec{
+		LocationType: "Slots",
+		Path:         "/yt/hdd2/slots",
+	}
+	testLocationImageCache = ytv1.LocationSpec{
+		LocationType: "ImageCache",
+		Path:         "/yt/hdd1/images",
+	}
+	testVolumeMounts = []corev1.VolumeMount{
+		{
+			Name:      "hdd1",
+			MountPath: "/yt/hdd1",
+		},
+	}
+	testVolumeClaimTemplates = []ytv1.EmbeddedPersistentVolumeClaim{
+		{
+			EmbeddedObjectMetadata: ytv1.EmbeddedObjectMetadata{
+				Name: "hdd1",
+			},
+			Spec: corev1.PersistentVolumeClaimSpec{
+				AccessModes:      []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce},
+				StorageClassName: &testStorageClassname,
+			},
+		},
+	}
+	testClusterNodeSpec = ytv1.ClusterNodesSpec{
+		Tags: []string{"rack:xn-a"},
+		Rack: "fake",
+	}
+)
+
+func getCommonSpec() ytv1.CommonSpec {
+	return ytv1.CommonSpec{
+		UseIPv6: true,
+	}
+}
+
+func getMasterConnectionSpec() ytv1.MasterConnectionSpec {
+	return ytv1.MasterConnectionSpec{
+		CellTag: 0,
+	}
+}
+
+func getYtsaurus() *ytv1.Ytsaurus {
```

Comment:
Please DO NOT COPY-PASTE such test assets. It will be impossible to maintain.

Reply:
@kruft can you please fix this tests with reusing CRD building from another package?